jsonpickle/ext/pandas.py

```diff
@@ -120,27 +116,102 @@ def restore(self, data):
 class PandasIndexHandler(BaseHandler):
     pp = PandasProcessor()
 
+    index_constructor = pd.Index
+    name_bundler = lambda _, obj: {'name': obj.name}
+
     def flatten(self, obj, data):
-        meta = {'dtype': str(obj.dtype), 'name': obj.name}
-        buf = json.dumps(obj.tolist())
+        name_bundle = self.name_bundler(obj)
+        meta = dict(dtype= str(obj.dtype), **name_bundle)
+        buf = encode(obj.tolist())
```

Comment:
This is an important difference.  The data of an index is now `encod`ed instead of just `json.dumps`.  This allows for the handling of more complex pandas types in indexes.

Reply:
Thanks for the comments explaining that bit, this looks great.